salt/client/ssh/__init__.py

```diff
@@ -464,19 +464,15 @@ def key_deploy(self, host, ret):
         """
         Deploy the SSH key if the minions don't auth
         """
+        # `or` initially was `and`, but was changed to be able to "deploy
+        # to multiple hosts" in #22661. Why? On each command error, this checks
+        # if a key deploy can be attempted without it being requested.
         if not isinstance(ret[host], dict) or self.opts.get("ssh_key_deploy"):
             target = self.targets[host]
             if target.get("passwd", False) or self.opts["ssh_passwd"]:
                 self._key_deploy_run(host, target, False)
-            return ret
```

Comment:
@Ch3LL need your review on this bit, I know you worked on it recently

Reply:
@Ch3LL just making sure you saw Pedro's request. :)